### ISSUE TITLE
splitSearchTerms control

### ISSUE THREAD

**Issue by danharrin** [laravel-filament/filament#13480]
### Discussed in https://github.com/filamentphp/filament/discussions/13466

<div type='discussions-op-text'>

<sup>Originally posted by **JohnRoux** July  4, 2024</sup>
There are a couple of things in Filament which are amazing, but don't really work well when you have a lot of data in a table.

Eg: If you have 10m+ records, doing the count(*) to get the total count of records can get reeeeeally slow, and it's not super intuitive as to how to fix that.

Additionally, searching with indexing is hard. But if you swap to `$search%` you can still get a benefit of at least 1 more level of index lookup.

This can make a pretty huge impact.

Currently I've got a `CustomListRecords` page which I use, it has simple options like:
```
    protected bool $simplePagination = false;
    protected bool $splitSearchTerms = true;
```

So all you need to do is change those flags on your ListRecords page, and you've swapped out.

Is this something that I should make a PR for? Super happy to if that's the case.

Either way, I'd love to contribute a bit towards some Filament at Scale documentation :+1: 

</div>

**Comment by danharrin** [comment]
@JohnRoux this has been implemented in v4 in #14965

**Comment by JohnRoux** [comment]
Champion 🏆💪 

Looks great!
